### PR TITLE
Safe url naming

### PR DESCRIPTION
**Pull request by mbogh**
My take on #146

Requires https://github.com/realm/jazzy-integration-specs/pull/4 to pass build

### CODE REVIEW

File: lib/jazzy/source_declaration.rb [hunk 1]
```diff
@@ -28,5 +30,9 @@ class SourceDeclaration
     def overview
       "#{abstract}\n\n#{discussion}".strip
     end
+
+    def filename
+      FilenameCleaner.sanitize("#{name}")
```
Comment (segiddins) [inline lib/jazzy/source_declaration.rb]: No need for string interpolation

**Comment by segiddins** [comment]
That gem seems to bring in a bunch of dependencies... Are we sure active support doesn't just have a method for this we could use?

**Comment by mbogh** [comment]
As far as I have read, there is no built-in way of escaping strings so they are safe for the filesystem.
"All" filename_cleaner does is applying some regexes.

``` ruby
url = "init(baseURL:sessionConfig:requestSerializer:responseSerializer:)"
url.gsub(/[^\w\.]/, '')
 => "initbaseURLsessionConfigrequestSerializerresponseSerializer"
```

**Comment by mbogh** [comment]
@segiddins do you have an opinion on the above?

**Comment by segiddins** [comment]
sure, using file-cleaner is fine if @jpsim is ok with it

**Comment by mbogh** [comment]
Seems that file-cleaner does not copy the input and hence changes the input.
``` ruby
head :001 > require "filename_cleaner"
 => true 
head :002 > name = "Global Variables"
 => "Global Variables" 
head :003 > new_name = FilenameCleaner.sanitize(name, '_', false)
 => "Global_Variables" 
head :004 > name
 => "Global_Variables"
```

I'm not that familiar with Ruby but it seems odd to me that I'm responsible for sending in a copy.

**Comment by jpsim** [comment]
Unless I'm missing something, it does seem like `URI::escape` is a preferable solution

**Pull request closed without merging**